core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -256,6 +258,7 @@ class GroupCoordinator(val brokerId: Int,
 
           case AwaitingSync =>
             group.get(memberId).awaitingSyncCallback = responseCallback
+            // FIXME: is below needed?
```

Comment:
Yeah, this was a note for myself which I meant to come back to since it didn't seem obvious when I read it why we needed it. The thing is that we won't expire the heartbeat as long as the SyncGroup request is stuck in purgatory (see the function `shouldKeepMemberAlive`), so while it may be reasonable to complete the current heartbeat, it doesn't make much sense to schedule the next one. Since we already call `completeAndScheduleNextHeartbeatExpiration` when the SyncGroup returns, I think we can remove this line and let the heartbeat expire.